source/Octopus.Client/Repositories/Async/BuiltInPackageRepositoryRepository.cs

```diff
@@ -62,18 +71,31 @@ public async Task<PackageFromBuiltInFeedResource> PushPackage(string fileName, S
                 Logger.Info("Pushing the complete package to the server, as delta compression was explicitly disabled");
             }
 
+            var link = await repository.Link("PackageUpload").ConfigureAwait(false);
+            object pathParameters;
+
+            // if the link doesn't contain overwritemode then we're connected to an older server, which uses the `replace` parameter  
+            if (link.Contains("overwritemode"))
+            {
+                pathParameters = new {replace = overwriteMode == OverwriteMode.OverwriteExisting};
```

Comment:
This logic looks opposite to what I'd expect, i.e. if server support `overwriteMode` we should pass in `overwriteMode` in the path params...

Reply:
Agree, most of these blocks have the `overwriteMode` case first, and this one should too.